Project 3
Jessica Kuang A14931276

Scientific Question: How does mortality risk associates with short sleeping duration?

Sleep duration is known to be associated with mortality rate. Many previous studies show that 7h is the optimal length of sleep, and sleeping for more than 8h or less than 6h both significantly increases all-cause mortality. In my study, I am trying to specifically study how significantly short sleeping duration influences mortality. The data I am using comes from the scientific article: Sleep duration, mortality and the influence of age. URL: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5680380/


Scientific Hypothesis: If the sleeping duration is shorter than the optimal length, then the risk of mortality increases

From the data, I am going to calculate the hazard ratio (HR) for each sleep duration group. Then, by using HR, I can use ttest to calculate the p-value to test if short sleep significantly increases mortality. I could not find raw data for my topic. The data I am using is manually imported into Python from the scientific article: Sleep duration, mortality and the influence of age. URL: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5680380/


Part 1: Load the Packages

Packages loaded include the following:
    Pandas: pandas provides many convenient data structures to allow easy analysis of real-world, large data
    SciPy: SciPy contains modules for many special functions, linear algebra, integration, and so on. scipy.stats which I am using contains a large number of statistical functions
    Operator: operator includes a set of efficient functions that are commonly used. truediv which I am using stands for true division

In [5]:
import pandas as pd
from scipy.stats import ttest_ind
from operator import truediv

Part 2: Load in the data

Data is imported manually as array first which then gets put into dataframe. 

In [6]:
#hours of sleep per night
hr = ['<=5', '6', '7', '>=8']
#total number of participants
N = [3270, 9192, 16822, 9907]
#total death
D = [486, 761, 1255, 1046]
#calculate death% for different hours of sleep/night. truediv is used so that each value in array D can be divided by corresponding value in array N
percentage = list(map(truediv, D, N)) 

#create dictionary and put the data into dataframe
deathdict = { 'hours/night':hr, 'number':N,'death':D, '%death':percentage }
death = pd.DataFrame(deathdict)

print(death)

#This is again secondary data from "Sleep duration, mortality and the influence of age". C stands for cause of death
C = ['CVD', 'cancer', 'suicide', 'respiratory_disease', 'neurological_disease', 'other' ]
#number of death corresponding to each cause
NC5 = [180, 180, 7, 21, 13, 85]
NC6 = [231, 364, 12, 24, 27, 103]
NC7 = [373, 532, 17, 56, 36, 141]
NC8 = [356, 469, 12, 31, 37, 141]

#create dictionary and put the data into dataframe
causedict = {'cause':C, 'number_of<=5h/night':NC5, 'number_of6h/night':NC6, 'number_of7h/night':NC7, 'number_of>=8h/night':NC8 }
cause = pd.DataFrame(causedict)

print(cause)

  hours/night  number  death    %death
0         <=5    3270    486  0.148624
1           6    9192    761  0.082789
2           7   16822   1255  0.074605
3         >=8    9907   1046  0.105582
                  cause  number_of<=5h/night  number_of6h/night  \
0                   CVD                  180                231   
1                cancer                  180                364   
2               suicide                    7                 12   
3   respiratory_disease                   21                 24   
4  neurological_disease                   13                 27   
5                 other                   85                103   

   number_of7h/night  number_of>=8h/night  
0                373                  356  
1                532                  469  
2                 17                   12  
3                 56                   31  
4                 36                   37  
5                141                  141  


Part 3: Performing bioinformatics analysis (p-value)

HR is calculated by comparing the percent death of each sleep duration group, the >=8h group is used as the standard. By calculating HR, we are then ready to calculate p-value using ttest.

In [7]:
#percentage of death for each cause for different sleeping durations
NC5percentage = [x/3270 for x in NC5]
NC6percentage = [x/9192 for x in NC6]
NC7percentage = [x/16822 for x in NC7]
NC8percentage = [x/9907 for x in NC8]

#from death%, using data of sleeping for 8h or more as standard, calculate hazard ratio of speepling for 5h or less, 6h, and 7h
NC5HR = list(map(truediv, NC5percentage, NC8percentage)) 
NC6HR = list(map(truediv, NC6percentage, NC8percentage)) 
NC7HR = list(map(truediv, NC7percentage, NC8percentage)) 

Part 4: Plotting the results

Ttest is used to generate p-value to test whether short sleep significantly increases mortality or not.

In [8]:
#ttest for hazard ratio of sleeping for 5/6h conpare to sleeping for 7h
ttest5 = ttest_ind(NC5HR, NC7HR)
ttest6 = ttest_ind(NC6HR, NC7HR)

print(ttest5)
print(ttest6)

Ttest_indResult(statistic=4.755507469944207, pvalue=0.0007737750747851186)
Ttest_indResult(statistic=1.199292231038599, pvalue=0.2580592223345451)


Part 5: Analysis of the results

The p value of comparing sleeping for 7h and sleeping for <=5h is 0.0008, less than 0.05, so sleeping for <=5h significantly increases all cause risk of death.
The p value of comparing sleeping for 7h and sleeping for 6h is 0.26, greater than 0.05, so sleeping for 6h does not significantly increase all cause risk of death.
My hyphothesis is partially correct that the risk of mortality indeed significantly increases when the sleeping duration is shorter or equal to 5h, while sleeping for 6h does not significantly increase risk of mortality even though it is also shorter than the optimal sleeping length of 7h.